adcc/testdata/qchem/generate_qchem_data.py

```diff
@@ -0,0 +1,171 @@
+pe                       {pe}
+ee_singlets              {singlet_states}
+ee_triplets              {triplet_states}
+input_bohr               {bohr}
+sym_ignore               true
+adc_davidson_maxiter     {maxiter}
+adc_davidson_conv        {conv_tol}
+adc_nguess_singles       {n_guesses}
+adc_davidson_maxsubspace {max_ss}
+adc_prop_es              true
+cc_rest_occ              {cc_rest_occ}
+
+! scf stuff
+use_libqints             true
+gen_scfman               true
+$end
+
+$molecule
+{charge} {multiplicity}
+{xyz}
+$end
+"""
+
+_pe_template = """
+$pe
+potfile {potfile}
+$end
+"""
+
+_method_dict = {
+    "adc0": "adc(0)",
+    "adc1": "adc(1)",
+    "adc2": "adc(2)",
+    "adc2x": "adc(2)-x",
+    "adc3": "adc(3)",
+    "cvs-adc0": "cvs-adc(0)",
+    "cvs-adc1": "cvs-adc(1)",
+    "cvs-adc2": "cvs-adc(2)",
+    "cvs-adc2x": "cvs-adc(2)-x",
+    "cvs-adc3": "cvs-adc(3)",
+}
+
+basis_remap = {
+    "sto3g": "sto-3g",
+    "def2tzvp": "def2-tzvp",
+    "ccpvdz": "cc-pvdz",
+}
+
+
+def clean_xyz(xyz):
+    return "\n".join(x.strip() for x in xyz.splitlines())
+
+
+def generate_qchem_input_file(fname, method, basis, coords, potfile=None,
+                              charge=0, multiplicity=1, memory=8000,
+                              singlet_states=5, triplet_states=0, bohr=True,
+                              maxiter=160, conv_tol=10, n_core_orbitals=0):
+    nguess_singles = 2 * max(singlet_states, triplet_states)
+    max_ss = 5 * nguess_singles
+    pe = potfile is not None
+    qci = _qchem_template.format(
+        method=_method_dict[method],
+        basis=basis_remap[basis],
+        memory=memory,
+        singlet_states=singlet_states,
+        triplet_states=triplet_states,
+        n_guesses=nguess_singles,
+        bohr=bohr,
+        maxiter=maxiter,
+        conv_tol=conv_tol,
+        max_ss=max_ss,
+        charge=charge,
+        multiplicity=multiplicity,
+        cc_rest_occ=n_core_orbitals,
+        xyz=clean_xyz(coords),
+        pe=pe
+    )
+    if pe:
+        qci += _pe_template.format(potfile=potfile)
+    qc_file = open(fname, "w")
+    qc_file.write(qci)
+    qc_file.close()
+
+
+def dump_qchem(molecule, method, basis, **kwargs):
+    with tempfile.TemporaryDirectory() as tmpdir:
+        pe = kwargs.get("potfile", None) is not None
+        if pe:
+            basename = f"{molecule}_{basis}_pe_{method}"
+        else:
+            basename = f"{molecule}_{basis}_{method}"
+        infile = os.path.join(tmpdir, basename + ".in")
+        outfile = os.path.join(tmpdir, basename + ".out")
+        geom = xyz[molecule].strip()
+        ret = {
+            "molecule": molecule,
+            "method": method,
+            "basis": basis,
+            "pe": pe,
+            "geometry": geom
+        }
+        generate_qchem_input_file(
+            infile, method, basis, geom, **kwargs
+        )
+        # only works with my (ms) fork of cclib
+        # github.com/maxscheurer/cclib, branch dev-qchem
```

Comment:
Why do you not make a PR there?

Reply:
I'd have to think about it. Some of the stuff could definitely go in there, but I also changed a lot of units that `cclib` stores its data in. Who wants excitation energies in cm^-1 ?! 😄